File: Problems/080-remove-duplicates-from-sorted-array-ii/remove-duplicates-from-sorted-array-ii.go
package problem080

func removeDuplicates(nums []int) int {
	if len(nums) <= 2 {
		return len(nums)
	}
	idx := 1
	for i := 2; i < len(nums); i++ {
		if nums[i] != nums[idx-1] {
			idx++
			nums[idx] = nums[i]
		}
	}
	return idx + 1
}

File: Problems/080-remove-duplicates-from-sorted-array-ii/remove-duplicates-from-sorted-array-ii.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        if len(nums) <= 2: return len(nums)

        idx = 1
        for i in range(2, len(nums)):
            if nums[i] != nums[idx-1]:
                idx += 1
                nums[idx] = nums[i]
                
        return idx+1